metrics-monitoring/README.adoc

```diff
@@ -0,0 +1,225 @@
+// Placeholder URL, when we get a HOST UI for the service we can put it here properly
+:service-url: https://console.redhat.com/application-services/streams/
+:registry-url: https://console.redhat.com/application-services/service-registry/
+:property-file-name: app-services.properties
+
+// Other upstream project names
+:samples-git-repo: https://github.com/redhat-developer/app-services-guides
+
+//URL components for cross refs
+:base-url: https://github.com/redhat-developer/app-services-guides/blob/main/
+:base-url-cli: https://github.com/redhat-developer/app-services-cli/tree/main/docs/
+:getting-started-url: getting-started/README.adoc
+:getting-started-service-registry-url: getting-started-service-registry/README.adoc
+:kafka-bin-scripts-url: kafka-bin-scripts/README.adoc
+:kafkacat-url: kafkacat/README.adoc
+:quarkus-url: quarkus/README.adoc
+:quarkus-service-registry-url: quarkus-service-registry/README.adoc
+:rhoas-cli-url: rhoas-cli/README.adoc
+:rhoas-cli-kafka-url: rhoas-cli-kafka/README.adoc
+:rhoas-cli-service-registry-url: rhoas-cli-service-registry/README.adoc
+:rhoas-cli-ref-url: commands
+:topic-config-url: topic-configuration/README.adoc
+:consumer-config-url: consumer-configuration/README.adoc
+:service-binding-url: service-discovery/README.adoc
+:access-mgmt-url: access-mgmt/README.adoc
+:access-mgmt-service-registry-url: access-mgmt-service-registry/README.adoc
+
+////
+END GENERATED ATTRIBUTES
+////
+
+[id="chap-monitoring-metrics"]
+= Monitoring metrics in {product-long}
+ifdef::context[:parent-context: {context}]
+:context: monitoring-metrics
+
+// Purpose statement for the assembly
+[role="_abstract"]
+As a developer or administrator, you can view metrics in {product} to visualize the performance and data usage for Kafka instances and topics that you have access to. You can view metrics directly in the {product} web console, or use the metrics API endpoint provided by {product} to import the data into your own metrics monitoring tool, such as Prometheus.
+
+//Additional line break to resolve mod docs generation error, not sure why. Leaving for now. (Stetson, 20 May 2021)
+
+[id="ref-supported-metrics_{context}"]
+== Supported metrics in {product}
+
+{product} supports the following metrics for Kafka instances and topics. In the {product} web console, a subset of these metrics is displayed in the *Dashboard* page of each Kafka instance.
+
+Broker metrics::
++
+--
+* `kafka_broker_quota_softlimitbytes`: Maximum amount of broker storage before producers are throttled
+
+* `kafka_broker_quota_totalstorageusedbytes`: Total storage used by the broker
+
+* `kafka_controller_kafkacontroller_global_partition_count`: Number of partitions across all topics in the cluster
+
+* `kafka_controller_kafkacontroller_offline_partitions_count`: Number of partitions without an active leader and that are therefore not writable or readable
+
+* `kafka_namespace:haproxy_server_bytes_in_total:rate5m`: *<@SME: Description?>*
```

Comment:
Ack. But for this and the outgoing rate, I've said "Incoming byte rate for **the broker** in the last five minutes" because all of these metrics seem to be specifically for "server" and "broker" settings, except the topic ones. Hence all other similar metrics refer to broker and are under a "Broker" heading. Also Confluent classifies all of these as metrics for the broker. So is it correct to say that for consistency? Not sure when I should say "Kafka instance" versus stick to saying "broker" for consistency and accuracy.

Reply:
it's not a placeholder but rather how the metric is grouped/aggregated. The metrics with `:` are recording rules and the naming convention was suggested by @david-martin

Reply:
Now it makes some sense, thanks! But the fact that they're implemented by recording rules is an impl detail as far as the user is concerned. I think it would be better to have a naming schema that hid that detail.

Reply:
@JameelB Ack. I see. I've just updated this and several other metric descriptions to say "in the instance" etc., instead of "in the broker", because there could be multiple brokers or other components in the instance involved for a metric. I think that broadens the scope a bit here and elsewhere to avoid the innacuracy.